File: app/java-ms/financial-accounting-service/src/main/java/com/example/accounting/port/in/CreateJournalUseCase.java
package com.example.accounting.port.in;

import com.example.accounting.domain.Journal;
import java.time.LocalDate;
import java.util.List;

public interface CreateJournalUseCase {
    Journal createJournal(LocalDate journalDate, String description, Integer fiscalYear, List<JournalEntryRequest> entries);
}

File: app/ruby/accounting_service/financial_accounting/app/ports/in/create_journal_use_case.rb
# frozen_string_literal: true

module Ports
  module In
    class CreateJournalUseCase
      # ユースケースのインターフェースを定義
      def execute(journal_date:, description:, fiscal_year:, entries:)
        raise NotImplementedError, "#{self.class}##{__method__} must be implemented"
      end
    end
  end
end
